packages/server/lib/util/agent.ts

```diff
@@ -0,0 +1,259 @@
+  httpsAgent: https.Agent
+
+  constructor (opts: http.AgentOptions = {}) {
+    opts.keepAlive = true
+    super(opts)
+    // we will need this if they wish to make http requests over an https proxy
+    this.httpsAgent = new https.Agent({ keepAlive: true })
+  }
+
+  createSocket (req, options, cb) {
+    if (process.env.HTTP_PROXY) {
+      const proxy = getProxyForUrl(options.href)
+
+      if (proxy) {
+        options.proxy = proxy
+
+        return this._createProxiedSocket(req, options, cb)
+      }
+    }
+
+    super.createSocket(req, options, cb)
+  }
+
+  _createProxiedSocket (req, options, cb) {
+    debug(`Creating proxied socket for ${options.href} through ${options.proxy}`)
+
+    const proxy = url.parse(options.proxy)
+
+    // set req.path to the full path so the proxy can resolve it
+    req.path = options.href
+
+    delete req._header // so we can set headers again
+
+    req.setHeader('host', `${options.host}:${options.port}`)
+    if (proxy.auth) {
+      req.setHeader('proxy-authorization', `basic ${Buffer.from(proxy.auth).toString('base64')}`)
+    }
+
+    // node has queued an HTTP message to be sent already, so we need to regenerate the
+    // queued message with the new path and headers
+    // https://github.com/TooTallNate/node-http-proxy-agent/blob/master/index.js#L93
+    regenerateRequestHead(req)
+
+    options.port = proxy.port
+    options.host = proxy.hostname
+    delete options.path // so the underlying net.connect doesn't default to IPC
+
+    if (proxy.protocol === 'https:') {
+      // gonna have to use the https module to reach the proxy, even though this is an http req
+      req.agent = this.httpsAgent
+
+      return this.httpsAgent.addRequest(req, options)
+    }
+
+    super.createSocket(req, options, cb)
+  }
+}
+
+class HttpsAgent extends https.Agent {
+  constructor (opts: https.AgentOptions = {}) {
+    opts.keepAlive = true
+    super(opts)
+  }
+
+  createConnection (options, cb) {
+    if (process.env.HTTPS_PROXY) {
+      const proxy = getProxyForUrl(options.href)
+
+      if (proxy) {
+        options.proxy = proxy
+
+        return this.createProxiedConnection(options, cb)
+      }
+    }
+
+    // @ts-ignore
+    cb(null, super.createConnection(options))
+  }
+
+  createProxiedConnection (options, cb) {
+    // heavily inspired by
+    // https://github.com/mknj/node-keepalive-proxy-agent/blob/master/index.js
+    debug(`Creating proxied socket for ${options.href} through ${options.proxy}`)
+
+    const proxy = url.parse(options.proxy)
+    const port = options.uri.port || 443
+    const hostname = options.uri.hostname
+
+    const proxySocket = createProxySock(proxy)
+
+    const onError = (err) => {
+      proxySocket.destroy()
+      cb(err)
+    }
+
+    let buffer = ''
+
+    const onData = (data) => {
+      debug(`Proxy socket for ${options.href} established`)
+
+      buffer += data.toString()
+
+      if (buffer.indexOf('\r\n\r\n') === -1) {
```

Comment:
you can use `String.prototype.includes` since it was introduced in node 6.x.x or just opt for lodash `_.includes`